File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        answer = [0] * len(temperatures)
        stack = []

        for idx, val in enumerate(temperatures):
            while stack and temperatures[stack[-1]] < val:
                poped_idx = stack.pop()
                answer[poped_idx] = idx - poped_idx 
            stack.append(idx)

        return answer

File: 0739-daily-temperatures/0739-daily-temperatures.kt
import java.util.Deque
import java.util.ArrayDeque

class Solution {
    fun dailyTemperatures(temperatures: IntArray): IntArray {
        val answer = IntArray(temperatures.size)
        val stack = ArrayDeque<Int>()
        
        for (i in 0 until temperatures.size){
            while (!stack.isEmpty() && temperatures[i] > temperatures[stack.peek()]){
                val last = stack.pop()
                answer[last] = i - last
            }
            stack.push(i)
        }
        return answer
    }
}